src/ethereum/crypto/elliptic_curve.py

```diff
@@ -71,3 +76,57 @@ def secp256k1_recover(r: U256, s: U256, v: U256, msg_hash: Hash32) -> Bytes:
 
     public_key = public_key.format(compressed=False)[1:]
     return public_key
+
+
+SECP256R1N = U256(
+    0xFFFFFFFF00000000FFFFFFFFFFFFFFFFBCE6FAADA7179E84F3B9CAC2FC632551
+)
+SECP256R1P = U256(
+    0xFFFFFFFF00000001000000000000000000000000FFFFFFFFFFFFFFFFFFFFFFFF
+)
+SECP256R1A = U256(
+    0xFFFFFFFF00000001000000000000000000000000FFFFFFFFFFFFFFFFFFFFFFFC
+)
+SECP256R1B = U256(
+    0x5AC635D8AA3A93E7B3EBBD55769886BC651D06B0CC53B0F63BCE3C3E27D2604B
+)
+
+
+def secp256r1_verify(
+    r: U256, s: U256, x: U256, y: U256, msg_hash: Hash32
+) -> None:
+    """
+    Verifies a P-256 signature.
+    Parameters
+    ----------
+    r :
+        the `r` component of the signature
+    s :
+        the `s` component of the signature
+    x:
+        the `x` coordinate of the public key
+    y:
+        the `y` coordinate of the public key
+    msg_hash :
+        Hash of the message being recovered.
+    Returns
+    -------
+    result : `ethereum.base_types.Bytes`
+        return 1 if the signature is valid, empty bytes otherwise
+    """
+    # Convert U256 to regular integers for DerSequence
+    r_int = int(r)
+    s_int = int(s)
+    x_int = int(x)
+    y_int = int(y)
+
+    sig = DerSequence([r_int, s_int]).encode()
+
+    try:
+        pubnum = ec.EllipticCurvePublicNumbers(x_int, y_int, ec.SECP256R1())
+        pubkey = pubnum.public_key(default_backend())
+        pubkey.verify(sig, msg_hash, ec.ECDSA(Prehashed(hashes.SHA256())))
+    except Exception as e:
+        raise Exception from e
```

Comment:
```suggestion
    pubnum = ec.EllipticCurvePublicNumbers(x_int, y_int, ec.SECP256R1())
    pubkey = pubnum.public_key(default_backend())
    pubkey.verify(sig, msg_hash, ec.ECDSA(Prehashed(hashes.SHA256())))
```

Reply:
Re-raising the same exception will turn the original exception into a generic one. `try : except` is not needed here unless we are doing something else